### ISSUE TITLE
Изменение свойств (ОС)

### ISSUE THREAD

**Issue by gsbelarus** [GoldenSoftwareLtd/gedemin#442]
Originally reported on Google Code with ID 442

```
Невозможно удалить документ 'ОС. Изменение свойств'. Пишет, что на данный
документ ссылаются записи. По основному оформлено 2 документа: Акт ввода в
эксплуатацию остатков и Изменение свойств.
```

Reported by `danilchyk` on 2007-03-05 13:14:05

**Comment by gsbelarus** [comment]
```
Пробовал на Эталоне, никаких проблем не увидел.
```

Reported by `flakekun` on 2007-03-21 10:27:43
- **Status changed**: `Invalid`

**Comment by gsbelarus** [comment]
```
Ситуации, когда определенная функциональность работает только на эталонной базе,
бывают  достаточно часто. Чтобы проверять ошибки надо держать какую-нибудь тестовую
базу и периодически загружать на нее свежие настройки из стартима. Ошибка эта есть.
Можно проверить на базе G:\Temp\bag\ETALON.FDB
```

Reported by `danilchyk` on 2007-03-21 11:14:32
- **Status changed**: `New`

**Comment by gsbelarus** [comment]
Reported by `flakekun` on 2007-03-26 07:47:27
- **Labels added**: GedeminExe

**Comment by gsbelarus** [comment]
Reported by `Alexander.GoldenSoft` on 2007-04-18 16:36:43

**Comment by gsbelarus** [comment]
```
Выглядит так, как будто позиция документа, движения и карточка удаляются не в том
порядке, в котором нужно. Если попробовать удалить позицию док-та из SQL-редактора,
появится исключение "Нарушение ссылочной целостности. Внешний ключ
"INV_FK_MOVEMENT_CARDK" в таблице "INV_MOVEMENT"". Если удалять отдельно движение,
проблем не возникает... Не думаю, что это связано с настройками. Если создать новый
док-т, с ним проблем не возникает
```

Reported by `alexandra.gsoftware` on 2007-04-27 12:10:31
- **Status changed**: `Accepted`

**Comment by gsbelarus** [comment]
```
Это связано точно не с настройками, такое возникает тогда когда по документу остаются
карточки, которых нет в движении. Правда описанная ситуация к этому не подходит. При
удалении документа сначала удаляется запись из пользовательской таблицы у которой на
триггере идет удаление из INV_movement, а затем удаляется запись из gd_document, и
блокировать это удаление может только свободно гуляющие карточки, которые зависят
друг от друга.
```

Reported by `mikleshoihet` on 2008-05-19 09:19:33

**Comment by gsbelarus** [comment]
```
А как эти карточки могли остаться "свободно гуляющими"?

Вы, кажется, предлагали тогда сделать обработку исключения, и, если не удалось
удалить все на триггерах, попытаться удалить сначала движение, как это сделали мы.

Этот баг все еще актуален. Когда на Паласе увеличится документооборот, он (баг)
вполне может повториться
```

Reported by `alexandra.gsoftware` on 2008-05-22 14:01:13

**Comment by gsbelarus** [comment]
```
Карточки не участвующие в движении могут оставться (движения не было, потом создали
при этом изменили признаки входящие признаки, при формирования движения выбралась
другая карточка старые карточки просто остались в базе). Да это баг, но красивого
(даже не красивого решения) я пока не вижу. (причем тут Палас, это характерно для
любой базы???)
```

Reported by `mikleshoihet` on 2008-05-22 14:38:35

**Comment by gsbelarus** [comment]
Reported by `gs1994` on 2009-03-07 19:10:19
- **Status changed**: `New`
- **Labels added**: Priority-Low, FA, Component-Logic
- **Labels removed**: Priority-Medium, ОС

**Comment by gsbelarus** [comment]
```
Вылезла на Витебске проблема с лишними карточками. Из-за этого отчеты по
аммортизации, да и вообще все отчеты, которые используют процедуру
USR$FA_P_GETINVCARDLIST выдают больше ОС'ов чем есть на самом деле.
```

Reported by `flakekun` on 2009-04-03 14:51:07
- **Labels added**: Priority-Medium
- **Labels removed**: Priority-Low

**Comment by gsbelarus** [comment]
Reported by `flakekun` on 2010-03-18 11:34:12
- **Status changed**: `Duplicate`
- **Merged into**: #1921
